### PR TITLE
support comercial license updates for more accurate data

### PR DESCRIPTION
**Pull request by face**
customize database udpate db script to support the commercial city database for accuracy

### CODE REVIEW

File: scripts/updatedb.js [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir)
+  }
+}
+
+// Ref: http://stackoverflow.com/questions/8493195/how-can-i-parse-a-csv-string-with-javascript
+// Return array of string values, or NULL if CSV string not well formed.
+// Return array of string values, or NULL if CSV string not well formed.
+
+function tryFixingLine (line) {
```
Comment (vigan-abd) [inline scripts/updatedb.js:83]: maybe we can use https://www.npmjs.com/package/csv-parse as it's well tested and is actively maintained? sorry for asking to remove this part, I know it took quite some time to figure it out :(

File: scripts/updatedb.js [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+          console.log('ERROR'.red + ': Could not retrieve checksum for', database.type, 'Aborting'.red)
+          console.log('Run with "force" to update without checksum')
+          client.abort()
+          process.exit()
+        }
+        cb(null, database)
+      })
+    }
+
+    const client = https.get(getHTTPOptions(checksumUrl), onResponse)
```
Comment (vigan-abd) [inline scripts/updatedb.js:199]: here I think we can use https://github.com/bitfinexcom/bfx-facs-http instead as it supports both async and cb support + http agents

File: scripts/updatedb.js [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+const fs = require('fs')
+const http = require('http')
+const https = require('https')
+const path = require('path')
+const zlib = require('zlib')
+
+fs.existsSync = fs.existsSync || path.existsSync
+
+const async = require('async')
+const colors = require('colors') // eslint-disable-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (vigan-abd) [reply scripts/updatedb.js:18]: and deps are missing in package.json:
```
Missing dependencies
* grenache-nodejs-http: ./example.js
* grenache-nodejs-link: ./example.js
* colors: ./scripts/updatedb.js
* iconv-lite: ./scripts/updatedb.js
* lazy: ./scripts/updatedb.js
* rimraf: ./scripts/updatedb.js
* yauzl: ./scripts/updatedb.js
* ip-address: ./scripts/updatedb.js
* https-proxy-agent: ./scripts/updatedb.js
```

File: scripts/updatedb.js [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+
+    tmpFilePipe.on('close', function () {
+      console.log(' DONE'.green)
+      cb(null, tmpFile, fileName, database)
+    })
+  }
+
+  mkdir(tmpFile)
+
+  const client = https.get(getHTTPOptions(downloadUrl), onResponse)
```
[the remark this answers is not in the document]
Comment (face) [reply scripts/updatedb.js:250]: per our discussion in slack, this is an copy of an upstream script we sill keep close to the upstream version.

File: scripts/updatedb.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (face) [reply scripts/updatedb.js:18]: Thanks, fixed.

**Pull request merged**